src/lib_json/json_reader.cpp

```diff
@@ -44,13 +45,13 @@
 #pragma warning(disable : 4996)
 #endif
 
-static int const stackLimit_g = 1000;
-#if __cplusplus >= 201103L
-thread_local static int stackDepth_g = 0;  // see readValue()
-#else
-static int       stackDepth_g = 0;  // see readValue()
+// Define JSONCPP_STACK_LIMIT as an appropriate integer at compile time to change the stack limit
+#if !defined(JSONCPP_STACK_LIMIT)
```

Comment:
I think this should be `JSONCPP_DEPRECATED_STACK_LIMIT`, since this should apply only to the old deprecated `Reader`, which nobody should be using anyway.